source/code/plugin/filter_cadvisor_health_container.rb

```diff
@@ -0,0 +1,263 @@
+        @@cluster_health_model_enabled = HealthMonitorUtils.is_cluster_health_model_enabled
+
+        def initialize
+            super
+            @cpu_capacity = 0.0
+            @memory_capacity = 0.0
+            @last_resource_refresh = DateTime.now.to_time.to_i
+            @metrics_to_collect_hash = {}
+        end
+
+        def configure(conf)
+            super
+            @log = HealthMonitorUtils.getLogHandle
+            @log.debug {'Starting filter_cadvisor2health plugin'}
+        end
+
+        def start
+            super
+            @metrics_to_collect_hash = HealthMonitorUtils.build_metrics_hash(@metrics_to_collect)
+            @log.debug "Calling ensure_cpu_memory_capacity_set cpu_capacity #{@cpu_capacity} memory_capacity #{@memory_capacity}"
+            node_capacity = HealthMonitorUtils.ensure_cpu_memory_capacity_set(@@hm_log, @cpu_capacity, @memory_capacity, @@hostName)
+            @cpu_capacity = node_capacity[0]
+            @memory_capacity = node_capacity[1]
+            @log.info "CPU Capacity #{@cpu_capacity} Memory Capacity #{@memory_capacity}"
+            #HealthMonitorUtils.refresh_kubernetes_api_data(@log, @@hostName)
+            @@health_monitor_config = HealthMonitorUtils.getHealthMonitorConfig
+            ApplicationInsightsUtility.sendCustomEvent("filter_cadvisor_health Plugin Start", {})
+        end
+
+        def filter_stream(tag, es)
+            if !@@cluster_health_model_enabled
+                @log.info "Cluster Health Model disabled in filter_cadvisor_health_container"
+                return []
+            end
+            new_es = MultiEventStream.new
+            #HealthMonitorUtils.refresh_kubernetes_api_data(@log, @hostName)
+            records_count = 0
+            es.each { |time, record|
+              begin
+                filtered_record = filter(tag, time, record)
+                if !filtered_record.nil?
+                    new_es.add(time, filtered_record)
+                    records_count += 1
+                end
+              rescue => e
+                router.emit_error_event(tag, time, record, e)
+              end
+            }
+            @log.debug "Filter Records Count #{records_count}"
+            new_es
+        end
+
+        def filter(tag, time, record)
+            begin
+                if record.key?("MonitorLabels")
+                    return record
+                end
+                object_name = record['DataItems'][0]['ObjectName']
+                counter_name = record['DataItems'][0]['Collections'][0]['CounterName'].downcase
+                if @metrics_to_collect_hash.key?(counter_name.downcase)
+                    metric_value = record['DataItems'][0]['Collections'][0]['Value']
+                    case object_name
+                    when @@object_name_k8s_container
+                        case counter_name.downcase
+                        when @@counter_name_cpu
+                            # @log.debug "Object Name #{object_name}"
```

Comment:
pls get rid of the comments if unused

Reply:
this file is unused. please ignore for this review